### PR TITLE
docs: add docs file for the peer exchange

### PR DESCRIPTION
**Pull request by cmwaters**
Adds a description to how the pex reactors work

Addresses: #6512

### CODE REVIEW

File: internal/p2p/pex/doc.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+Package PEX (Peer exchange) handles all the logic necessary for nodes to share information about their peers to other nodes.
```
Comment (alexanderbez) [inline internal/p2p/pex/doc.go:2]: I typically wrap my markdown at 80 characters, but 120 also suffices.

File: internal/p2p/pex/doc.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Specifically, this is the exchange of addresses that a peer can use to discover more peers within the network.
+
+The PEX reactor is a continuous service which periodically requests addresses and serves addresses to other peers. 
+There are two versions of this service aligning with the two p2p frameworks that Tendermint currently has.
```
Comment (alexanderbez) [inline internal/p2p/pex/doc.go:6]:
```suggestion
There are two versions of this service aligning with the two p2p frameworks that Tendermint currently supports.
```

File: internal/p2p/pex/doc.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+V1 is coupled with the Switch (whcih handles peer connections and routing of messages) and, alongside exhanging
```
Comment (alexanderbez) [inline internal/p2p/pex/doc.go:8]:
```suggestion
V1 is coupled with the Switch (which handles peer connections and routing of messages) and, alongside exchanging 
```

File: internal/p2p/pex/doc.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+node has a sufficient amount of peers connected.
+
+V2 is embedded with the new p2p stack and uses the peer manager to advertise peers as well as add new peers to the peer
+store. The V2 reactor passes a different set of proto messages which include a list of [urls](https://golang.org/pkg/net/url/#URL).
+These can be used to save a set of endpoints that each peer uses. The V2 reactor has backwards compatibility with V1. It can
+also handle V1 messages.
+
+The V2 reactor is able to tweak the intensity of it's search by decreasing or increasing the interval between each request. 
+It tracks connected peers via a linked list, sending a request to the node at the front of the list and adding it to the 
+back of the list once a response is received. Using this method, a node is able to spread out the load of requesting peers accross
```
Comment (alexanderbez) [inline internal/p2p/pex/doc.go:19]:
```suggestion
back of the list once a response is received. Using this method, a node is able to spread out the load of requesting peers across
```

**Review by alexanderbez — CHANGES_REQUESTED** [review]

File: internal/p2p/pex/doc.go (hunk 1 above)
Comment (cmwaters) [reply internal/p2p/pex/doc.go:2]: Do we want to standardise this? I can set my editor to 80

File: internal/p2p/pex/doc.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+peers as well as add new peers to the peer store. The V2 reactor passes a
+different set of proto messages which include a list of
+[urls](https://golang.org/pkg/net/url/#URL).These can be used to save a set of
+endpoints that each peer uses. The V2 reactor has backwards compatibility with
+V1. It can also handle V1 messages.
+
+The V2 reactor is able to tweak the intensity of it's search by decreasing or
+increasing the interval between each request. It tracks connected peers via a
+linked list, sending a request to the node at the front of the list and adding
+it to the back of the list once a response is received. Using this method, a
```
Comment (williambanfield) [inline internal/p2p/pex/doc.go:25]: > It tracks connected peers via a linked list, sending a request to the node at the front of the list and adding it to the back of the list once a response is received. 

What types of messages does it do this for? All types?

Comment (cmwaters) [reply internal/p2p/pex/doc.go:25]: Yeah it sends either V1 or V2 depending on what type of pex reactor it thinks the peer is running

**Comment by cmwaters** [comment]
Do you mind unblocking this @alexanderbez when you get the chance (unless there's something else here I'm missing)? thanks

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**